src/d3a/d3a_core/redis_connections/aggregator_connection.py

```diff
@@ -43,32 +37,47 @@ def is_controlling_device(self, device_uuid):
 
     def _add_batch_event(self, device_uuid, event, batch_event_dict):
         aggregator_uuid = self.device_aggregator_mapping[device_uuid]
+        create_subdict_or_update(batch_event_dict, aggregator_uuid, event)
+
+    def _delete_not_owned_devices_from_dict(self, area_stats_tree_dict):
+        out_dict = deepcopy(area_stats_tree_dict)
+        self._delete_not_owned_devices(area_stats_tree_dict, out_dict)
+        return out_dict
 
-        if aggregator_uuid not in batch_event_dict:
-            batch_event_dict[aggregator_uuid] = []
+    def _delete_not_owned_devices(self, indict, outdict):
+        for area_uuid, area_dict in indict.items():
+            if 'children' in area_dict:
+                self._delete_not_owned_devices(indict[area_uuid]['children'],
+                                               outdict[area_uuid]['children'])
+            else:
+                if area_uuid not in self.device_aggregator_mapping:
+                    outdict[area_uuid] = {'area_name': area_dict['area_name']}
```

Comment:
I did not get how exactly you're deleting the not owned devices

Reply:
Normally the `outdict[area_uuid]` would contain a lot of information, but here we only expose the name.